cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureStorageAccountService.java

```diff
@@ -40,13 +45,22 @@
     @Inject
     private AzureStorage armStorage;
 
+    @Inject
+    private AzureCloudResourceService azureCloudResourceService;
+
+    @Inject
+    private ResourceNotifier resourceNotifier;
+
     public void createStorageAccount(AuthenticatedContext ac, AzureClient client, String resourceGroup, String storageName, String region, CloudStack stack) {
         StorageAccount storageAccount = client.getStorageAccountByGroup(resourceGroup, storageName);
         if (storageAccount == null) {
             try {
                 LOGGER.info("Creating storage account: {}", storageName);
-                armStorage.createStorage(client, storageName, LOCALLY_REDUNDANT, resourceGroup, region, isEncryptionNeeded(stack), stack.getTags());
+                StorageAccount storage = armStorage.createStorage(client, storageName, LOCALLY_REDUNDANT, resourceGroup, region, isEncryptionNeeded(stack),
+                        stack.getTags());
                 pollStorageAccountCreation(ac, new StorageAccountCheckerContext(client, resourceGroup, storageName));
+                CloudResource cloudResource = azureCloudResourceService.buildCloudResource(storage.name(), storage.id(), ResourceType.AZURE_STORAGE);
+                azureCloudResourceService.saveCloudResources(resourceNotifier, ac.getCloudContext(), List.of(cloudResource));
```

Comment:
I would put this into a finally clause. Since the storage account has more than 1 resource it could happen that the storage account is created but for some reason the blobService not, and an exception is thrown.

Reply:
You are right, but we do not know the storage acc id in the "finally" clause, we could go and fetch the deployment from Azure there but it is cumbersome as we don't have the deployment's id.
in such cases, the whole deployment will fail, the user will be notified about the failing storage account and at most a storage account will not be deleted, which is free. As this is a temp solution, I would not make a major effort to fix this.